pyam/iiasa.py

```diff
@@ -119,27 +168,24 @@ def __init__(self, name=None, creds=None, auth_url=_AUTH_URL):
         self._auth_url = auth_url  # scenario services manager API
         self._base_url = None  # database connection API
 
-        self._token, self._user = _get_token(creds, auth_url=self._auth_url)
+        self.auth = SceSeAuth(creds=creds, auth_url=self._auth_url)
 
         # connect if provided a name
         self._connected = None
         if name:
             self.connect(name)
 
-        if self._user:
-            logger.info(f"You are connected as user `{self._user}`")
+        if self.auth.user is not None:
+            logger.info(f"You are connected as user `{self.auth.user}`")
         else:
             logger.info("You are connected as an anonymous user")
 
-    @property
-    def _headers(self):
-        return {"Authorization": f"Bearer {self._token}"}
-
     @property
     @lru_cache()
     def _connection_map(self):
-        url = "/".join([self._auth_url, "applications"])
-        r = requests.get(url, headers=self._headers)
+        # TODO: application-list will be reimplemented in conjunction with ixmp-server
+        url = "/".join([self._auth_url, "legacy", "applications"])
+        r = requests.get(url, headers=self.auth())
```

Comment:
I would use httpx here as well. You can pass the auth object when instantiating the client and it will be called for every request.

Reply:
Thanks, implemented!